Pod/Classes/PINAnimatedImageManager.m

```diff
@@ -113,7 +129,7 @@ - (void)animatedPathForImageData:(NSData *)animatedImageData infoCompletion:(PIN
   __block BOOL startProcessing = NO;
   __block PINSharedAnimatedImage *sharedAnimatedImage = nil;
   {
-    [_lock lockWithBlock:^{
+    [_lock lockWhenCondition:PINAnimatedImageManagerConditionReady];
```

Comment:
`animatedPathForImageData:infoCompletion:completion:` can be called on the main thread. You should probably dispatch to a background queue before locking.